lark/exceptions.py

```diff
@@ -108,12 +101,25 @@ def match_examples(self, parse_fn, examples, token_type_match_fallback=False, us
 
                         except AttributeError:
                             pass
-                        if not candidate[0]:
+                        if candidate[0] is None:
                             logger.debug("Same State match at example [%s][%s]" % (i, j))
                             candidate = label, False
 
         return candidate[0]
 
+class UnexpectedEOF(ParseError, UnexpectedInput):
```

Comment:
Are we throwing this exception for Lark?

Reply:
What do you mean?

Reply:
Are we using this exception when Lalr reached the end of its input, or only earley?

Reply:
Only for Earley, right now LALR throws UnexpectedToken instead. We could change it so that both throw it on EOF, or neither one. Maybe it's even a good idea. But it seems outside the scope of this PR.